AssertionError where "data" is unexpectedly null when a file contains only triple slash references

**A.d.ts**

```ts
declare class Test {
}
```

**test.ts**

```ts
/// <reference path="./A.d.ts" />
```

**Output**

```
> deno run test.ts
Check file:///V:/dprint-plugins/test.ts
error: AssertionError: "data" is unexpectedly null for "file:///V:/dprint-plugins/A.d.ts".
    at assert (deno:cli/tsc/99_main_compiler.js:68:13)
    at Object.getSourceFile (deno:cli/tsc/99_main_compiler.js:286:7)
    at findSourceFileWorker (deno:cli/tsc/00_typescript.js:111060:29)
    at findSourceFile (deno:cli/tsc/00_typescript.js:110970:26)
    at deno:cli/tsc/00_typescript.js:110922:85
    at getSourceFileFromReferenceWorker (deno:cli/tsc/00_typescript.js:110889:34)
    at processSourceFile (deno:cli/tsc/00_typescript.js:110922:13)
    at deno:cli/tsc/00_typescript.js:111208:17
    at Object.forEach (deno:cli/tsc/00_typescript.js:384:30)
    at processReferencedFiles (deno:cli/tsc/00_typescript.js:111207:16)
```

Deno 1.11.3

I had a more complex scenario and will maybe open a separate issue for that if I can figure out a more minimal example. The above is ok once adding a statement to test.ts.

I'll just put my other same error in here with a non-empty file

**types/deploy.d.ts**

```ts
/// <reference path="./deploy.fetchevent.d.ts" />
```

**types/deploy.fetchevent.d.ts**

```ts
// Copyright 2018-2021 Deno Land Inc. All rights reserved. MIT license.

// deno-lint-ignore-file

/// <reference no-default-lib="true" />
/// <reference lib="esnext" />

declare class FetchEvent extends Event {
    request: Request;
    respondWith(response: Response | Promise<Response>): void;
}
```

**test.tsx**

```ts
/// <reference no-default-lib="true"/>
/// <reference lib="esnext" />
/// <reference path="./types/deploy.d.ts" />

/** @jsx h */
/** @jsxFrag Fragment */
import { h } from "https://x.lcas.dev/preact@10.5.12/mod.js";

function App() {
    return (
        <html>
            <head>
                <title>Hello from JSX</title>
            </head>
            <body>
                <h1>Hello world</h1>
            </body>
        </html>
    );
}
```

**Output**

```
> deno run test.tsx
Check file:///V:/dprint-plugins/test.tsx
error: AssertionError: "data" is unexpectedly null for "file:///V:/dprint-plugins/types/deploy.fetchevent.d.ts".
    at assert (deno:cli/tsc/99_main_compiler.js:68:13)
    at Object.getSourceFile (deno:cli/tsc/99_main_compiler.js:286:7)
    at findSourceFileWorker (deno:cli/tsc/00_typescript.js:111060:29)
    at findSourceFile (deno:cli/tsc/00_typescript.js:110970:26)
    at deno:cli/tsc/00_typescript.js:110922:85
    at getSourceFileFromReferenceWorker (deno:cli/tsc/00_typescript.js:110889:34)
    at processSourceFile (deno:cli/tsc/00_typescript.js:110922:13)
    at deno:cli/tsc/00_typescript.js:111208:17
    at Object.forEach (deno:cli/tsc/00_typescript.js:384:30)
    at processReferencedFiles (deno:cli/tsc/00_typescript.js:111207:16)
```

--- a/cli/tests/info/data_null_error/data_null_error.out
+++ b/cli/tests/info/data_null_error/data_null_error.out
@@ -1,0 +1,6 @@
+local: [WILDCARD]mod.ts
+type: TypeScript
+dependencies: 1 unique (total [WILDCARD])
+
+file://[WILDCARD]/mod.ts ([WILDCARD])
+└── file://[WILDCARD]/types.d.ts ([WILDCARD])

--- a/cli/tests/info/data_null_error/mod.ts
+++ b/cli/tests/info/data_null_error/mod.ts
@@ -1,0 +1,1 @@
+/// <reference path="./types.d.ts" />

--- a/cli/tests/info/data_null_error/types.d.ts
+++ b/cli/tests/info/data_null_error/types.d.ts
@@ -1,0 +1,1 @@
+declare class Test {}

--- a/cli/tests/integration/info_tests.rs
+++ b/cli/tests/integration/info_tests.rs
@@ -111,3 +111,10 @@ itest!(_054_info_local_imports {
   output: "054_info_local_imports.out",
   exit_code: 0,
 });
+
+// Tests for AssertionError where "data" is unexpectedly null when
+// a file contains only triple slash references (#11196)
+itest!(data_null_error {
+  args: "info info/data_null_error/mod.ts",
+  output: "info/data_null_error/data_null_error.out",
+});

